pkg/kv/kvserver/raftlog/iterator.go

```diff
@@ -90,44 +91,35 @@ func (it *Iterator) Close() {
 }
 
 func (it *Iterator) load() (bool, error) {
-	ok, err := it.iter.Valid()
-	if err != nil || !ok {
+	if ok, err := it.iter.Valid(); err != nil || !ok {
 		return false, err
 	}
-
-	it.entry, err = NewEntryFromRawValue(it.iter.UnsafeValue())
-	if err != nil {
+	if err := raftEntryFromRawValue(it.iter.UnsafeValue(), &it.entry); err != nil {
```

Comment:
do you want to explicitly reset `it.entry` here? Otherwise one might wonder whether callers that are holding on to the last `Entry()` can cause data races. I think it's not (gogoprot unmarshal code never reuses slices contained in the struct) but still.

Reply:
Done. I wonder if simply returning it from `raftEntryFromRawValue` would be better.

Reply:
Yes, better to return `raftpb.Entry` from this function.

Reply:
Done.